<a href="https://colab.research.google.com/github/Shameem-Yuseph/DSA_2025/blob/main/ETL_case_study.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [4]:
import pandas as pd

In [5]:
import numpy as np

In [6]:
data = pd.read_csv("/content/all_month.csv")
#Load the dataset into a DataFrame
data = pd.DataFrame(data)

In [7]:
data

time   latitude   longitude    depth   mag magType  \
0     2025-09-22T07:53:49.230Z  33.815333 -116.591333  14.5400  0.69      ml   
1     2025-09-22T07:50:25.360Z  40.298332 -124.436668  10.1200  2.44      md   
2     2025-09-22T07:42:43.580Z  33.536167 -116.716500   5.5900  0.52      ml   
3     2025-09-22T07:22:10.205Z  62.220600 -150.331900  14.8000  1.40      ml   
4     2025-09-22T07:00:50.918Z  31.330000 -103.680000   7.8430  2.90      ml   
...                        ...        ...         ...      ...   ...     ...   
9807  2025-08-23T08:33:13.760Z  33.462000 -116.603833  10.5000  1.16      ml   
9808  2025-08-23T08:33:00.733Z  28.756000  -99.103000   6.0425  1.90      ml   
9809  2025-08-23T08:26:38.460Z  54.755833 -163.982167   0.9100  0.00      ml   
9810  2025-08-23T08:20:04.440Z  58.058500 -157.221700  21.2000  1.50      ml   
9811  2025-08-23T08:15:37.260Z  35.832333 -121.322167   8.2100  1.77      md   

       nst    gap     dmin   rms  ...                   updated  \
0     36.0   44.0  0.03525  0.12  ...  2025-09-22T07:57:19.931Z   
1     21.0  264.0  0.07263  0.10  ...  2025-09-22T07:57:19.833Z   
2     32.0   29.0  0.03394  0.12  ...  2025-09-22T07:46:18.991Z   
3      NaN    NaN      NaN  0.44  ...  2025-09-22T07:23:52.103Z   
4     18.0   50.0  0.10000  0.10  ...  2025-09-22T07:16:11.116Z   
...    ...    ...      ...   ...  ...                       ...   
9807  71.0   24.0  0.03264  0.21  ...  2025-08-24T13:24:26.370Z   
9808  27.0   62.0  0.10000  0.40  ...  2025-08-25T12:48:32.744Z   
9809   5.0   90.0  0.04728  0.18  ...  2025-08-24T05:29:14.470Z   
9810   NaN    NaN      NaN  0.68  ...  2025-08-23T08:21:55.833Z   
9811  37.0  156.0  0.09657  0.07  ...  2025-08-30T02:52:19.895Z   

                                   place        type horizontalError  \
0             6 km W of Palm Springs, CA  earthquake        0.150000   
1                13 km W of Petrolia, CA  earthquake        1.560000   
2                   4 km WSW of Anza, CA  earthquake        0.130000   
3     11 km SSW of Trapper Creek, Alaska  earthquake             NaN   
4                10 km E of Toyah, Texas  earthquake        0.821146   
...                                  ...         ...             ...   
9807               12 km SSE of Anza, CA  earthquake        0.150000   
9808           9 km NE of Hilltop, Texas  earthquake        2.261349   
9809     38 km WSW of False Pass, Alaska  earthquake        0.400000   
9810         19 km SSE of Egegik, Alaska  earthquake             NaN   
9811         24 km NNW of San Simeon, CA  earthquake        0.230000   

     depthError  magError  magNst     status  locationSource magSource  
0      0.340000  0.156000    16.0  automatic              ci        ci  
1      0.520000  0.130000    18.0  automatic              nc        nc  
2      0.330000  0.131000    17.0  automatic              ci        ci  
3      0.200000       NaN     NaN  automatic              ak        ak  
4      1.463623  0.100000    24.0   reviewed              tx        tx  
...         ...       ...     ...        ...             ...       ...  
9807   0.290000  0.165000    26.0   reviewed              ci        ci  
9808   1.277091  0.100000    18.0   reviewed              tx        tx  
9809   0.610000  0.304161     5.0   reviewed              av        av  
9810   0.300000       NaN     NaN  automatic              ak        ak  
9811   0.500000  0.151000    27.0   reviewed              nc        nc  

[9812 rows x 22 columns]

In [8]:
number_of_records =len(data)
number_of_records

9812

In [9]:
unique_location = data['place'].nunique()
unique_location

5218

In [10]:
data.head(5)

time   latitude   longitude   depth   mag magType  \
0  2025-09-22T07:53:49.230Z  33.815333 -116.591333  14.540  0.69      ml   
1  2025-09-22T07:50:25.360Z  40.298332 -124.436668  10.120  2.44      md   
2  2025-09-22T07:42:43.580Z  33.536167 -116.716500   5.590  0.52      ml   
3  2025-09-22T07:22:10.205Z  62.220600 -150.331900  14.800  1.40      ml   
4  2025-09-22T07:00:50.918Z  31.330000 -103.680000   7.843  2.90      ml   

    nst    gap     dmin   rms  ...                   updated  \
0  36.0   44.0  0.03525  0.12  ...  2025-09-22T07:57:19.931Z   
1  21.0  264.0  0.07263  0.10  ...  2025-09-22T07:57:19.833Z   
2  32.0   29.0  0.03394  0.12  ...  2025-09-22T07:46:18.991Z   
3   NaN    NaN      NaN  0.44  ...  2025-09-22T07:23:52.103Z   
4  18.0   50.0  0.10000  0.10  ...  2025-09-22T07:16:11.116Z   

                                place        type horizontalError depthError  \
0          6 km W of Palm Springs, CA  earthquake        0.150000   0.340000   
1             13 km W of Petrolia, CA  earthquake        1.560000   0.520000   
2                4 km WSW of Anza, CA  earthquake        0.130000   0.330000   
3  11 km SSW of Trapper Creek, Alaska  earthquake             NaN   0.200000   
4             10 km E of Toyah, Texas  earthquake        0.821146   1.463623   

   magError  magNst     status  locationSource magSource  
0     0.156    16.0  automatic              ci        ci  
1     0.130    18.0  automatic              nc        nc  
2     0.131    17.0  automatic              ci        ci  
3       NaN     NaN  automatic              ak        ak  
4     0.100    24.0   reviewed              tx        tx  

[5 rows x 22 columns]

In [11]:
data.columns

Index(['time', 'latitude', 'longitude', 'depth', 'mag', 'magType', 'nst',
       'gap', 'dmin', 'rms', 'net', 'id', 'updated', 'place', 'type',
       'horizontalError', 'depthError', 'magError', 'magNst', 'status',
       'locationSource', 'magSource'],
      dtype='object')

In [12]:
#convert time to datetime

In [13]:
data['time']=pd.to_datetime(data['time'])
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9812 entries, 0 to 9811
Data columns (total 22 columns):
 #   Column           Non-Null Count  Dtype              
---  ------           --------------  -----              
 0   time             9812 non-null   datetime64[ns, UTC]
 1   latitude         9812 non-null   float64            
 2   longitude        9812 non-null   float64            
 3   depth            9812 non-null   float64            
 4   mag              9811 non-null   float64            
 5   magType          9811 non-null   object             
 6   nst              8681 non-null   float64            
 7   gap              8681 non-null   float64            
 8   dmin             8681 non-null   float64            
 9   rms              9811 non-null   float64            
 10  net              9812 non-null   object             
 11  id               9812 non-null   object             
 12  updated          9812 non-null   object             
 13  place            9

In [14]:
#Drop records with missing latitude, longitude, mag

In [15]:
print(data.isnull().sum())

time                  0
latitude              0
longitude             0
depth                 0
mag                   1
magType               1
nst                1131
gap                1131
dmin               1131
rms                   1
net                   0
id                    0
updated               0
place                 0
type                  0
horizontalError    2002
depthError            2
magError           1171
magNst             1150
status                0
locationSource        0
magSource             0
dtype: int64


In [16]:
data = data.dropna(subset=['latitude','longitude','mag'])

In [18]:
# Filter only earthquakes with magnitude >= 4.0
data[data['mag']>=4.0]

time  latitude  longitude    depth  mag  \
15   2025-09-22 05:46:06.309000+00:00  -22.6455   170.4294   10.000  4.9   
22   2025-09-22 04:29:24.262000+00:00   52.4528   160.0421   50.131  4.8   
25   2025-09-22 03:59:07.220000+00:00   11.9746   142.2153   13.291  5.0   
38   2025-09-22 02:45:44.035000+00:00  -15.4776   -70.5599  197.751  4.3   
50   2025-09-22 01:20:01.115000+00:00   40.2851    24.0985   12.194  4.9   
...                               ...       ...        ...      ...  ...   
9773 2025-08-23 10:56:58.401000+00:00   51.6223   160.0907   10.000  4.3   
9781 2025-08-23 10:22:11.035000+00:00   36.0353   138.9697  129.454  4.1   
9787 2025-08-23 09:36:46.944000+00:00   42.7785   146.4886   23.236  4.4   
9794 2025-08-23 09:14:22.080000+00:00   13.0343   -90.3959   17.000  6.0   
9795 2025-08-23 09:07:59.056000+00:00  -32.4859   -71.4911   47.457  4.5   

     magType   nst    gap   dmin   rms  ...                   updated  \
15        mb  38.0  104.0  2.756  0.92  ...  2025-09-22T07:19:51.040Z   
22        mb  58.0  162.0  1.021  0.97  ...  2025-09-22T04:57:57.040Z   
25        mb  56.0   54.0  3.045  0.78  ...  2025-09-22T06:09:15.215Z   
38        mb  18.0  139.0  2.341  0.99  ...  2025-09-22T03:01:44.040Z   
50       mww  46.0   57.0  1.246  0.71  ...  2025-09-22T08:05:36.319Z   
...      ...   ...    ...    ...   ...  ...                       ...   
9773      mb  38.0  127.0  1.659  0.88  ...  2025-09-19T12:25:01.040Z   
9781      mb  41.0  136.0  0.796  1.11  ...  2025-09-19T10:11:38.040Z   
9787      mb  43.0  136.0  3.137  0.56  ...  2025-09-19T09:50:02.040Z   
9794     mww  80.0  120.0  1.050  1.52  ...  2025-09-19T12:25:59.130Z   
9795     mwr  50.0  130.0  0.551  0.53  ...  2025-09-19T12:03:03.963Z   

                                              place        type  \
15              289 km ESE of Tadine, New Caledonia  earthquake   
22    117 km SE of Petropavlovsk-Kamchatsky, Russia  earthquake   
25                     south of the Mariana Islands  earthquake   
38                   24 km NNE of Santa Lucía, Peru  earthquake   
50                11 km ESE of Ouranoupolis, Greece  earthquake   
...                                             ...         ...   
9773               185 km SE of Vilyuchinsk, Russia  earthquake   
9781                       3 km WNW of Ogano, Japan  earthquake   
9787                      95 km SE of Nemuro, Japan  earthquake   
9794              87 km SW of Acajutla, El Salvador  earthquake   
9795                     24 km W of La Ligua, Chile  earthquake   

     horizontalError depthError  magError  magNst    status  locationSource  \
15              7.79      1.443     0.101    31.0  reviewed              us   
22             10.49      7.762     0.062    80.0  reviewed              us   
25              8.96      4.620     0.066    72.0  reviewed              us   
38             13.10      6.738     0.119    20.0  reviewed              us   
50              5.13      5.187     0.053    34.0  reviewed              us   
...              ...        ...       ...     ...       ...             ...   
9773            8.13      1.895     0.093    33.0  reviewed              us   
9781            9.61      6.501     0.097    29.0  reviewed              us   
9787            5.49      5.715     0.101    28.0  reviewed              us   
9794            6.35      1.870     0.032    95.0  reviewed              us   
9795            3.33      4.554     0.083    14.0  reviewed              us   

     magSource  
15          us  
22          us  
25          us  
38          us  
50          us  
...        ...  
9773        us  
9781        us  
9787        us  
9794        us  
9795        us  

[852 rows x 22 columns]

In [19]:
#Add a column day_of_week from time
data['day_of_week'] =data['time'].dt.day_name()
data.columns

/tmp/ipython-input-3200885488.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['day_of_week'] =data['time'].dt.day_name()


Index(['time', 'latitude', 'longitude', 'depth', 'mag', 'magType', 'nst',
       'gap', 'dmin', 'rms', 'net', 'id', 'updated', 'place', 'type',
       'horizontalError', 'depthError', 'magError', 'magNst', 'status',
       'locationSource', 'magSource', 'day_of_week'],
      dtype='object')

In [20]:
conditions =[
    (data['mag']<4.0),
    (data['mag']>=4.0)&(data['mag']<=6.0),
    (data['mag']>6.0)
]
choices=['Low','Moderate','High']
data['severity_level']=np.select(conditions,choices,default='unknown')

/tmp/ipython-input-2218767570.py:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['severity_level']=np.select(conditions,choices,default='unknown')


In [21]:
data.columns

Index(['time', 'latitude', 'longitude', 'depth', 'mag', 'magType', 'nst',
       'gap', 'dmin', 'rms', 'net', 'id', 'updated', 'place', 'type',
       'horizontalError', 'depthError', 'magError', 'magNst', 'status',
       'locationSource', 'magSource', 'day_of_week', 'severity_level'],
      dtype='object')

In [22]:
#count number of earthquakes per place

In [23]:
earthquake_count=data['place'].value_counts()
print(earthquake_count)

place
15 km NE of Milford, Utah                  202
27 km W of Hebgen Lake Estates, Montana     82
7 km NNW of The Geysers, CA                 58
7 km WNW of Cobb, CA                        57
47 km ESE of Beatty, Nevada                 54
                                          ... 
48 km NNW of Paxson, Alaska                  1
289 km ESE of Tadine, New Caledonia          1
8 km WSW of Calimesa, CA                     1
36 km N of Stevens Village, Alaska           1
34 km NE of Lucerne Valley, CA               1
Name: count, Length: 5217, dtype: int64


In [24]:
#Compute average magnitude and max depth per day
daily_stats = data.groupby('day_of_week').agg(
    average_magnitude=('mag', 'mean'),
    max_depth=('depth', 'max')
)
display(daily_stats)

average_magnitude  max_depth
day_of_week                              
Friday                1.465409    564.825
Monday                1.410836    639.511
Saturday              1.504574    593.972
Sunday                1.458911    554.091
Thursday              1.491480    392.952
Tuesday               1.432679    608.312
Wednesday             1.352362    569.035

In [25]:
data.to_csv('cleaned_earthquakes.csv')
daily_stats.to_csv('earthquake_summary.csv')

In [26]:
import sqlite3

# Create a SQLite database file and connect to it
conn = sqlite3.connect('earthquakes.db')
cursor = conn.cursor()

In [27]:
# Store cleaned dataset into a SQLite table
df =pd.read_csv('/content/cleaned_earthquakes.csv')
df.to_sql('earthquakes', conn, index=False,if_exists='replace')

9811

Open In Colab

import pandas as pd
import numpy as np
     

data =pd.read_csv("/content/all_month.csv")
#Load the dataset into a DataFrame
data = pd.DataFrame(data)
     

data
     
time	latitude	longitude	depth	mag	magType	nst	gap	dmin	rms	...	updated	place	type	horizontalError	depthError	magError	magNst	status	locationSource	magSource
0	2025-09-20T15:59:20.245Z	39.199700	-119.608500	3.0000	1.00	ml	10.0	109.46	0.101000	0.0462	...	2025-09-20T16:01:51.702Z	4 km SSW of Dayton, Nevada	earthquake	NaN	0.800000	0.550000	6.0	automatic	nn	nn
1	2025-09-20T15:53:42.262Z	60.310500	-152.131500	77.9000	1.80	ml	NaN	NaN	NaN	0.4000	...	2025-09-20T15:55:14.828Z	38 km NW of Ninilchik, Alaska	earthquake	NaN	0.500000	NaN	NaN	automatic	ak	ak
2	2025-09-20T15:48:54.143Z	62.897200	-147.936900	250.0000	2.00	ml	NaN	NaN	NaN	0.8800	...	2025-09-20T15:55:17.612Z	75 km SE of Cantwell, Alaska	earthquake	NaN	7.500000	NaN	NaN	automatic	ak	ak
3	2025-09-20T15:47:58.757Z	32.351000	-102.075000	4.9533	1.10	ml	10.0	64.00	0.000000	0.3000	...	2025-09-20T16:00:12.191Z	36 km NW of Stanton, Texas	earthquake	0.00	1.467807	0.300000	6.0	automatic	tx	tx
4	2025-09-20T15:41:48.425Z	-23.438600	-68.417700	101.4060	4.50	mb	59.0	92.00	0.532000	0.8600	...	2025-09-20T16:04:59.981Z	62 km SSW of San Pedro de Atacama, Chile	earthquake	6.27	5.448000	0.075000	56.0	reviewed	us	us
...	...	...	...	...	...	...	...	...	...	...	...	...	...	...	...	...	...	...	...	...	...
9988	2025-08-21T16:57:59.190Z	38.812333	-122.757000	1.2400	0.21	md	13.0	93.00	0.011880	0.0300	...	2025-08-29T21:32:17.560Z	3 km WSW of Cobb, CA	earthquake	0.24	0.340000	0.419000	11.0	reviewed	nc	nc
9989	2025-08-21T16:43:27.010Z	53.849500	-166.859167	4.6300	0.00	ml	7.0	119.00	0.057290	0.1000	...	2025-08-22T03:43:58.970Z	21 km WSW of Dutch Harbor, Alaska	earthquake	0.38	1.120000	0.239543	7.0	reviewed	av	av
9990	2025-08-21T16:39:12.040Z	46.719667	-121.117833	8.8600	0.90	ml	13.0	139.00	0.202000	0.1700	...	2025-08-21T20:21:30.650Z	17 km SW of Nile, Washington	earthquake	0.59	11.920000	0.111201	9.0	reviewed	uw	uw
9991	2025-08-21T16:30:52.940Z	47.421000	-122.790667	5.5300	1.30	ml	12.0	126.00	0.046370	0.4900	...	2025-09-05T00:49:30.050Z	4 km SE of Belfair, Washington	earthquake	1.14	2.560000	0.135646	4.0	reviewed	uw	uw
9992	2025-08-21T16:29:43.860Z	38.839333	-122.825500	1.6500	0.58	md	8.0	101.00	0.007834	0.0100	...	2025-08-21T16:31:19.667Z	9 km WNW of Cobb, CA	earthquake	0.72	2.310000	0.350000	8.0	automatic	nc	nc
9993 rows × 22 columns



#Identify number of records and unique locations
number_of_records =len(data)
number_of_records

     
9993

unique_location = data['place'].nunique()
unique_location
     
5263

#Print top 5 rows and column names

data.head(5)


     
time	latitude	longitude	depth	mag	magType	nst	gap	dmin	rms	...	updated	place	type	horizontalError	depthError	magError	magNst	status	locationSource	magSource
0	2025-09-20T15:59:20.245Z	39.1997	-119.6085	3.0000	1.0	ml	10.0	109.46	0.101	0.0462	...	2025-09-20T16:01:51.702Z	4 km SSW of Dayton, Nevada	earthquake	NaN	0.800000	0.550	6.0	automatic	nn	nn
1	2025-09-20T15:53:42.262Z	60.3105	-152.1315	77.9000	1.8	ml	NaN	NaN	NaN	0.4000	...	2025-09-20T15:55:14.828Z	38 km NW of Ninilchik, Alaska	earthquake	NaN	0.500000	NaN	NaN	automatic	ak	ak
2	2025-09-20T15:48:54.143Z	62.8972	-147.9369	250.0000	2.0	ml	NaN	NaN	NaN	0.8800	...	2025-09-20T15:55:17.612Z	75 km SE of Cantwell, Alaska	earthquake	NaN	7.500000	NaN	NaN	automatic	ak	ak
3	2025-09-20T15:47:58.757Z	32.3510	-102.0750	4.9533	1.1	ml	10.0	64.00	0.000	0.3000	...	2025-09-20T16:00:12.191Z	36 km NW of Stanton, Texas	earthquake	0.00	1.467807	0.300	6.0	automatic	tx	tx
4	2025-09-20T15:41:48.425Z	-23.4386	-68.4177	101.4060	4.5	mb	59.0	92.00	0.532	0.8600	...	2025-09-20T16:04:59.981Z	62 km SSW of San Pedro de Atacama, Chile	earthquake	6.27	5.448000	0.075	56.0	reviewed	us	us
5 rows × 22 columns


data.columns
     
Index(['time', 'latitude', 'longitude', 'depth', 'mag', 'magType', 'nst',
       'gap', 'dmin', 'rms', 'net', 'id', 'updated', 'place', 'type',
       'horizontalError', 'depthError', 'magError', 'magNst', 'status',
       'locationSource', 'magSource'],
      dtype='object')

#Convert time to datetime
     


data['time']=pd.to_datetime(data['time'])
data.info()
     
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9993 entries, 0 to 9992
Data columns (total 22 columns):
 #   Column           Non-Null Count  Dtype              
---  ------           --------------  -----              
 0   time             9993 non-null   datetime64[ns, UTC]
 1   latitude         9993 non-null   float64            
 2   longitude        9993 non-null   float64            
 3   depth            9993 non-null   float64            
 4   mag              9992 non-null   float64            
 5   magType          9992 non-null   object             
 6   nst              8861 non-null   float64            
 7   gap              8861 non-null   float64            
 8   dmin             8861 non-null   float64            
 9   rms              9992 non-null   float64            
 10  net              9993 non-null   object             
 11  id               9993 non-null   object             
 12  updated          9993 non-null   object             
 13  place            9993 non-null   object             
 14  type             9993 non-null   object             
 15  horizontalError  7975 non-null   float64            
 16  depthError       9993 non-null   float64            
 17  magError         8823 non-null   float64            
 18  magNst           8843 non-null   float64            
 19  status           9993 non-null   object             
 20  locationSource   9993 non-null   object             
 21  magSource        9993 non-null   object             
dtypes: datetime64[ns, UTC](1), float64(12), object(9)
memory usage: 1.7+ MB

 #Drop records with missing latitude, longitude, mag
     


print(data.isnull().sum())
     
time                  0
latitude              0
longitude             0
depth                 0
mag                   1
magType               1
nst                1132
gap                1132
dmin               1132
rms                   1
net                   0
id                    0
updated               0
place                 0
type                  0
horizontalError    2018
depthError            0
magError           1170
magNst             1150
status                0
locationSource        0
magSource             0
dtype: int64

data = data.dropna(subset=['latitude','longitude','mag'])

     

# Filter only earthquakes with magnitude >= 4.0
data[data['mag'] >= 4.0]


     
time	latitude	longitude	depth	mag	magType	nst	gap	dmin	rms	...	updated	place	type	horizontalError	depthError	magError	magNst	status	locationSource	magSource
4	2025-09-20 15:41:48.425000+00:00	-23.4386	-68.4177	101.406	4.5	mb	59.0	92.0	0.532	0.86	...	2025-09-20T16:04:59.981Z	62 km SSW of San Pedro de Atacama, Chile	earthquake	6.27	5.448	0.075	56.0	reviewed	us	us
5	2025-09-20 15:31:09.338000+00:00	-58.1161	-24.6544	10.000	5.2	mb	75.0	48.0	7.633	0.73	...	2025-09-20T15:49:54.040Z	South Sandwich Islands region	earthquake	8.57	1.862	0.061	89.0	reviewed	us	us
17	2025-09-20 14:06:44.193000+00:00	-7.1365	155.8194	95.225	4.9	mb	32.0	139.0	4.668	0.54	...	2025-09-20T14:25:40.040Z	97 km SSE of Panguna, Papua New Guinea	earthquake	5.88	11.451	0.061	83.0	reviewed	us	us
18	2025-09-20 14:06:08.600000+00:00	52.9290	160.5458	10.000	4.5	mb	35.0	204.0	1.150	0.65	...	2025-09-20T14:50:39.040Z	129 km E of Petropavlovsk-Kamchatsky, Russia	earthquake	13.25	1.972	0.068	64.0	reviewed	us	us
20	2025-09-20 14:02:27.796000+00:00	-6.8605	156.0656	99.026	4.1	mb	17.0	179.0	4.618	0.90	...	2025-09-20T14:57:22.040Z	86 km SE of Kieta, Papua New Guinea	earthquake	13.25	20.952	0.145	15.0	reviewed	us	us
...	...	...	...	...	...	...	...	...	...	...	...	...	...	...	...	...	...	...	...	...	...
9956	2025-08-21 18:57:22.226000+00:00	53.6258	161.2762	36.294	4.7	mb	68.0	134.0	1.687	0.61	...	2025-09-16T14:50:53.040Z	187 km ENE of Petropavlovsk-Kamchatsky, Russia	earthquake	11.37	7.317	0.062	79.0	reviewed	us	us
9959	2025-08-21 18:51:23.463000+00:00	28.0083	139.3748	511.333	4.3	mb	78.0	64.0	2.655	0.62	...	2025-09-16T14:53:17.040Z	Bonin Islands, Japan region	earthquake	11.19	8.411	0.039	186.0	reviewed	us	us
9960	2025-08-21 18:44:25.147000+00:00	50.1719	159.0046	10.000	4.1	mb	33.0	138.0	2.864	0.52	...	2025-09-16T16:16:11.040Z	212 km ESE of Severo-Kuril’sk, Russia	earthquake	12.10	1.930	0.109	23.0	reviewed	us	us
9976	2025-08-21 17:16:32.193000+00:00	51.9710	160.5043	10.000	4.9	mb	66.0	125.0	1.548	0.86	...	2025-09-16T15:07:21.040Z	176 km SE of Petropavlovsk-Kamchatsky, Russia	earthquake	9.12	1.878	0.070	64.0	reviewed	us	us
9977	2025-08-21 17:16:20.665000+00:00	52.0814	160.3966	10.000	4.9	mb	92.0	114.0	1.424	0.70	...	2025-09-16T15:07:36.040Z	162 km SE of Petropavlovsk-Kamchatsky, Russia	earthquake	8.28	1.814	0.040	200.0	reviewed	us	us
848 rows × 22 columns


#Add a column day_of_week from time
data['day_of_week'] =data['time'].dt.day_name()
data.columns

     
Index(['time', 'latitude', 'longitude', 'depth', 'mag', 'magType', 'nst',
       'gap', 'dmin', 'rms', 'net', 'id', 'updated', 'place', 'type',
       'horizontalError', 'depthError', 'magError', 'magNst', 'status',
       'locationSource', 'magSource', 'day_of_week'],
      dtype='object')
Create a column severity_level based on magnitude:

○ <4.0: "Low"

○ 4.0 - 6.0: "Moderate"

○ 6.0+: "High"


conditions =[
    (data['mag']<4.0),
    (data['mag']>=4.0)&(data['mag']<=6.0),
    (data['mag']>6.0)
]
choices=['Low','Moderate','High']
data['severity_level']=np.select(conditions,choices,default='unknown')
     

data.columns
     
Index(['time', 'latitude', 'longitude', 'depth', 'mag', 'magType', 'nst',
       'gap', 'dmin', 'rms', 'net', 'id', 'updated', 'place', 'type',
       'horizontalError', 'depthError', 'magError', 'magNst', 'status',
       'locationSource', 'magSource', 'day_of_week', 'severity_level'],
      dtype='object')

#count number of earthquakes per place

earthquake_count=data['place'].value_counts()
print(earthquake_count)

     
place
15 km NE of Milford, Utah                  203
27 km W of Hebgen Lake Estates, Montana     81
7 km WNW of Cobb, CA                        59
7 km NNW of The Geysers, CA                 58
6 km WNW of Cobb, CA                        55
                                          ...
198 km SSE of Vilyuchinsk, Russia            1
29 km W of Salamatof, Alaska                 1
46 km E of Pedro Bay, Alaska                 1
208 km ESE of Ozernovskiy, Russia            1
3 km E of Big Bear Lake, CA                  1
Name: count, Length: 5262, dtype: int64






     

#Compute average magnitude and max depth per day
daily_stats = data.groupby('day_of_week').agg(
    average_magnitude=('mag', 'mean'),
    max_depth=('depth', 'max')
)
display(daily_stats)
     
average_magnitude	max_depth
day_of_week
Friday	1.469079	564.825
Monday	1.387024	639.511
Saturday	1.475806	593.972
Sunday	1.382038	554.091
Thursday	1.503625	511.333
Tuesday	1.410940	612.812
Wednesday	1.353294	569.035

data.to_csv('cleaned_earthquakes.csv')
daily_stats.to_csv('earthquake_summary.csv')
     

import sqlite3

# Create a SQLite database file and connect to it
conn = sqlite3.connect('earthquakes.db')
cursor = conn.cursor()
     

# Store cleaned dataset into a SQLite table
df =pd.read_csv('/content/cleaned_earthquakes.csv')
df.to_sql('earthquakes', conn, index=False,if_exists='replace')


     
9992
Create a flowchart of your ETL logic
EXTRACT DATA
Load Data from CSV
TRANSFORM DATA
{Convert time to datetime};
 -->{Handle Missing Values};
 -->{Filter Earthquakes by Magnitude};
 --> {Add day_of_week Column};
 --> {Add severity_level Column};
 --> {Count Earthquakes per Place};
 --> {Compute Daily Statistics};
 --> [Save to CSV];
 --> [Save to CSV];
LOAD DATA
 --> [Connect to SQLite Database];
 --> [Store Data in SQLite Table];
 --> [Store Daily Stats in SQLite Table];
A short explanation of your process
Extraction: I loaded the earthquake data from the provided CSV file into a pandas DataFrame.

Transformation: I performed several data cleaning and feature engineering steps: Converted the 'time' column to datetime objects. Dropped rows with missing values in critical columns ('latitude', 'longitude', 'mag'). Filtered for earthquakes with a magnitude of 4.0 or greater Created a 'day_of_week' column from the timestamp. Created a 'severity_level' column based on the magnitude. Counted the number of earthquakes per place. Computed the average magnitude and maximum depth for each day of the week.

Loading: I saved the cleaned dataset and the daily statistics to CSV files and also stored them in tables within a SQLite database.



     
One insight your team discovered from the data
Based on the daily_stats table, one insight is that earthquakes, on average, tend to have slightly higher magnitudes on Thursdays and Saturdays compared to other days, while the maximum recorded depth of earthquakes varies across days, with Monday showing the highest maximum depth in this dataset.And 15 km NE of Milford, Utah has a frequent occurence of earthquake